### ISSUE TITLE
[net7.0-iOS] Cannot debug my project with Linker disabled

### ISSUE THREAD

**Issue by DDHSchmidt** [dotnet/maui#15703]
### Description

Whenever I try to start my application on Visual Studio for Mac to deploy it onto my iPad (16.5.0), the build runs just fine, but immediately on starting, the app crashes with the following message:

`System.ArgumentOutOfRangeException)	"Token 0x2001417 is not valid in the scope of module Microsoft.iOS.dll (Parameter 'metadataToken')"`

It doesn't occurr when I start a simple, blank MAUI app on my Ipad.

Interestingly my app throws a *different* error when I try t start it up on the simulator:
`System.TypeLoadException: VTable setup of type MyApp.Controllers.ApplicationController failed`

By pure chance I discovered that when I change the "Linker behaviour" to "Link Framework SDKs only" it magically enables my app to start flawlessly on my iPad *and* on the simulator!

That, of course, disables HotReload for me and I have to restart the app over and over for every little change I want to make...
Can someone give me a hint what could be causing this? I found next to nothing about this in the issues, Google or Bing.

### Steps to Reproduce

1. Compile project in debug mode with "Linker behaviour" set to "Don't link"
2. Let it start on physical device
3. Observe crash on start up

### Link to public reproduction project repository

na

### Version with bug

7.0.59

### Last version that worked well

Unknown/Other

### Affected platforms

iOS

### Affected platform versions

iOS 16.5, VS for Mac 17.5.7 (build 6), macOS 13.4

### Did you find any workaround?

Set Linker behaviour to "Link Framework SDKs only" and loose HotReload

### Relevant log output

```shell
//Real device
System.ArgumentOutOfRangeException)	"Token 0x2001417 is not valid in the scope of module Microsoft.iOS.dll (Parameter 'metadataToken')"	

   at System.Reflection.RuntimeModule.ResolveType(Module , IntPtr , Int32 , Type[] , Type[] )
   at System.Reflection.RuntimeModule.ResolveType(Int32 , Type[] , Type[] )
   at System.Reflection.Module.ResolveType(Int32 )
   at ObjCRuntime.Class.ResolveToken(Module , UInt32 )
   at ObjCRuntime.Class.ResolveFullTokenReference(UInt32 )
   at ObjCRuntime.Class.ResolveTokenReference(UInt32 , UInt32 )
   at ObjCRuntime.Class.ResolveTypeTokenReference(UInt32 )
   at ObjCRuntime.Class.FindType(NativeHandle , Boolean& )
   at ObjCRuntime.Class.Lookup(IntPtr , Boolean )
   at ObjCRuntime.Class.Lookup(IntPtr )
   at ObjCRuntime.Runtime.GetNSObject[NSBundle](IntPtr )
   at Foundation.NSBundle.get_MainBundle()
   at Microsoft.Maui.Platform.ElementExtensions.HasSceneManifest(IUIApplicationDelegate platformApplication)
   at Microsoft.Maui.MauiUIApplicationDelegate.RespondsToSelector(Selector sel)
   at UIKit.UIApplication.UIApplicationMain(Int32 , String[] , IntPtr , IntPtr )
   at UIKit.UIApplication.Main(String[] , Type , Type )
   at MyApp.Program.Main(String[] args) in /Users/hschm/MyApp/MyApp/Platforms/iOS/Program.cs:line 14

//Simulator
System.TypeLoadException: VTable setup of type MyApp.Controllers.ApplicationController failed

   at MyApp.AppDelegate.CreateMauiApp() in /Users/hschm/MyApp/MyApp/Platforms/iOS/AppDelegate.cs:line 24
   at Microsoft.Maui.MauiUIApplicationDelegate.WillFinishLaunching(UIApplication application, NSDictionary launchOptions)
   at UIKit.UIApplication.UIApplicationMain(Int32 argc, String[] argv, IntPtr principalClassName, IntPtr delegateClassName)
   at UIKit.UIApplication.Main(String[] args, Type principalClass, Type delegateClass)
   at MyApp.Program.Main(String[] args) in /Users/hschm/MyApp/MyApp/Platforms/iOS/Program.cs:line 14
```

**Comment by drasticactions** [comment]
If you need the linker enabled to debug your application (and you don't need it from a new template MAUI application), that implies that you are using either some library that requires it or doing something within your application that needs it. We would need more information about what you're doing with your app (and hopefully a reproduction project with whatever you're doing) to be able to to see why you would need it enabled.

CC @rolfbjarne, do you have any ideas on what to look for here?

**Comment by DDHSchmidt** [comment]
I get where you're coming from but I'm under a NDA for this project so I can't just upload it or the corporate lawyers would sue me til kingdom come...

I was hoping for a tip like "Switch on this verbose debug log option here and it will show you which NuGet dependency is crashing the linker".

Alternatively, I just tried to add all of my NuGet-dependencies to a blank MAUI project, but that worked just fine. I guess I have to actually *use* the dependencies somewhere in my code to trigger the linker failure...

Otherwise the only option left is to play whack-a-mole with my 30 dependencies and outcomment each dependency, outcomment the hundreds of now non-compiling lines, then rebuild the project and see if the linker crash is gone now...

Please tell me that is not my only option left :(

**Comment by PrestigiousF** [comment]
Ran across this error today. I did a lot of enabling and disabling of packages, but nothing... I then remembered something as simple as deleting the bin and obj folder usually fixes a lot of untraceable issues.

Here's the order that I did this to solve this mysterious issue:

Clean Project > Delete bin & obj folders > Restore Nuget Packages > Build/Debug

I hope this helps anyone that has this issue in the future.
